### ISSUE TITLE
F821 false positive: `from .foo import Bar` doesn't mark `foo` as defined.

### ISSUE THREAD

**Issue by alona-em** [astral-sh/ruff#23233]
### Summary

*`spam/__init__.py`*

```python
from .foo import Bar

print(foo.Bar)
```

*`spam/foo.py`*
```python
class Bar:
    pass
```

`ruff` things this is an error.

```
% uvx ruff check ./spam/__init__.py --select=F821
F821 Undefined name `foo`
 --> spam/__init__.py:3:7
  |
1 | from .foo import Bar
2 |
3 | print(foo.Bar)
  |       ^^^
  |

Found 1 error.
```

But it works fine:

```
% python3 -c 'import spam'
<class 'spam.foo.Bar'>
```

This works due to weird python behavior (https://docs.python.org/3/reference/import.html#submodules). 

(See also https://github.com/microsoft/pyright/issues/782, where pyright used to have this same bug).

### Version

ruff 0.15.0 (ce5f7b612 2026-02-03)

**Comment by ntBre** [comment]
Thanks for the report! This seems closely related to the F401 changes in https://github.com/astral-sh/ruff/pull/20200 and many of the linked issues, but it seems that this hasn't come up for F821, at least based on my searches.
